packages/core/src/tracing/spanHandle.js

```diff
@@ -194,6 +203,22 @@ function _annotateWithArray(target, path, value) {
   }
 }
 
+/**
+ * Marks the span as an error (that is, it sets the error count for the span to 1). You can optionally provide a boolean
+ * attribute: if the provided value is true, the span will be marked as an error (just as if no argument had been
+ * provided), if it is false, the span will be marked as not being an error (that is, the error count will be set to 0).
+ * If no argument is provided, the default behavior is to mark the span as an error.
+ *
+ * @param {boolean?} error whether to mark the span as an error
+ */
+SpanHandle.prototype.markAsError = function markAsError(error = true) {
```

Comment:
TBD: This change provides a helper function for customers to easily add an error count (instead of directly using `span.ec = 1`). But what about the **error message**? Wondering if we want to extend the fn design. Without an error message this feels a little incomplete?

Reply:
> instead of directly using span.ec = 1

To clarify, this would only work like this:
```
const span = instana.currentSpan();
// get a reference to the internal span object
const internalSpan = span.span;
internalSpan..ec = 1;
```

So it is not so much about making it easier, but more about having a clear separation between our public API and our internal data structure. If we recommend to users to work on `instana.currentSpan().span` directly, we can no longer change our internal format easily in the future. Of course, users can still do that, but then they use internals and if that breaks, it is not our responsibilty.

Back to the error message suggestion: The error message is a normal annotation in span.data. It also is expected in different paths, depending on the type of span (that is, `span.data.http.error` vs. `span.data.mysql.error` vs `span.data.lambda.error` etc.) 

So from my perspective there are two reasons against adding this to the API here:

* Users can already use `span.annotate` to add the error message (if they know what kind of span they are dealing with and where the error message needs to go).
* On the API level here, it would be somewhat cumbersome for ourselves to figure out whether we would need to write `span.data.http.error` or `span.data.mysql.error` or `span.data.lambda.error` or whatever.

Reply:
> Users can already use span.annotate to add the error message

Yes true but that is complicated, don't you think?

> On the API level here, it would be somewhat cumbersome for ourselves to figure out whether we would need to write

I thought we usually have a single key in `span.data`? Is that not correct?

We could add something like this:

```
markAsError(true, 'error message')
```

```
function markAsError(bool, msg, path) {
   if (msg) {
     const path = path || tryToGetPath(this.span.data)
     
     if (path && path.length) _annotateWithString(this.span.data, path, msg);
     else logger.warn('Cannot set error message on span. Please annotate the error message using the annotate function.');
   }
   
   ...
 }
 
 function tryToGetPath(data) {
   const keys = Object.keys(data)
   if (keys.length && keys.length === 1) return keys[0]
   return false;
}
```

Reply:
Sold! I thought about this a bit more, and I think you are right. I'll start working on this and update the PR.